mmdet3d/models/roi_heads/bbox_heads/multi_group_head.py

```diff
@@ -0,0 +1,880 @@
+        """Loss function for CenterHead.
+
+        Args:
+            gt_bboxes_3d (list[:obj:`LiDARInstance3DBoxes`]): Ground
+                truth gt boxes.
+            gt_labels_3d (list[torch.Tensor]): Labels of boxes.
+            preds_dicts (dict): Output of forward function.
+
+        Returns:
+            dict: Computed losses.
+
+                - task0.loss_heatmap (torch.Tensor): Loss of heatmap for task0.
+                - task0.loss_bbox (torch.Tensor): Loss of location for task0.
+                - task1.loss_heatmap (torch.Tensor): Loss of heatmap for task1.
+                - task1.loss_bbox (torch.Tensor): Loss of location for task1.
+                - task2.loss_heatmap (torch.Tensor): Loss of heatmap for task2.
+                - task2.loss_bbox (torch.Tensor): Loss of location for task2.
+                - task3.loss_heatmap (torch.Tensor): Loss of heatmap for task3.
+                - task3.loss_bbox (torch.Tensor): Loss of location for task3.
+                - task4.loss_heatmap (torch.Tensor): Loss of heatmap for task4.
+                - task4.loss_bbox (torch.Tensor): Loss of location for task4.
+                - task5.loss_heatmap (torch.Tensor): Loss of heatmap for task5.
+                - task5.loss_bbox (torch.Tensor): Loss of location for task5.
+        """
+        heatmaps, anno_boxes, inds, masks = self.get_targets(
+            gt_bboxes_3d, gt_labels_3d)
+        loss_dict = dict()
+        for task_id, preds_dict in enumerate(preds_dicts):
+            # heatmap focal loss
+            preds_dict[0]['heatmap'] = clip_sigmoid(preds_dict[0]['heatmap'])
+            num_pos = heatmaps[task_id].eq(1).float().sum().item()
+            loss_heatmap = self.loss_cls(
+                preds_dict[0]['heatmap'],
+                heatmaps[task_id],
+                avg_factor=max(num_pos, 1))
+            target_box = anno_boxes[task_id]
+            # reconstruct the anno_box from multiple reg heads
+            preds_dict[0]['anno_box'] = torch.cat(
+                (preds_dict[0]['reg'], preds_dict[0]['height'],
+                 preds_dict[0]['dim'], preds_dict[0]['vel'],
+                 preds_dict[0]['rot']),
+                dim=1)
+
+            # Regression loss for dimension, offset, height, rotation
+            ind = inds[task_id]
+            num = masks[task_id].float().sum()
+            pred = preds_dict[0]['anno_box'].permute(0, 2, 3, 1).contiguous()
+            pred = pred.view(pred.size(0), -1, pred.size(3))
+            pred = self._gather_feat(pred, ind)
+            mask = masks[task_id].unsqueeze(2).expand_as(target_box).float()
+            isnotnan = (~torch.isnan(target_box)).float()
+            mask *= isnotnan
+            box_loss = torch.sum(self.loss_reg(pred, target_box, mask), 1) / (
+                num + 1e-4)
+            code_weights = self.train_cfg.get('code_weights', [])
+            loss_bbox = (box_loss * box_loss.new_tensor(code_weights)).sum()
```

Comment:
Pass code_weights as an argument to the line 642.
See an example [here](https://github.com/open-mmlab/mmdetection3d/blob/master/mmdet3d/models/dense_heads/anchor3d_head.py#L220)